### PR TITLE
Fixed an inefficient nearestNode and nearestLink implementation where…

### PR DESCRIPTION
**Pull request by bobpappas**
In target.js, a full sorted array, the full size of the nodes or links was being spliced. Instead, only an array is kept sorted for the expected count of nodes or links.

For a 20k set of nodes and many links, this was taking 2+ minutes with the Chrome performance regression of Array.splice: https://bugs.chromium.org/p/chromium/issues/detail?id=835558

With this optimization, the Chrome performance regression is not a factor.

### CODE REVIEW

**Comment by bobpappas** [comment]
DuoJS is not working correctly for development in grapher-target. @cixzhang removed component.json from grapher which I think is the culprit. The workaround is to change grapher-target/build/build.js to: Grapher = require('../components/ayasdi-grapher@v1.3.2/build/grapher.js');

This is after DuoJS installs Grapher in it's components directory.

**Comment by cixzhang** [comment]
Looks like Chrome fix is on 68.0.3418.0.

Can we add a comment with link to chromium bug?

We may as well test the performance (timestamps?) in the spec as well.

**Comment by cixzhang** [comment]
Oh and don't forget to build :)

**Comment by cixzhang** [comment]
Merge #7 to fix test issues relating to duo.

**Comment by bobpappas** [comment]
Bummer, Travis CI build is having trouble. I can't reproduce it locally by following the Node 4.1.0 Travis CI steps. It works fine.

```
>> Error caught from PhantomJS. More info can be found by opening the Spec Runner in a browser.
Warning: SyntaxError: Parse error Use --force to continue.
```
@cixzhang, any suggestions?

**Comment by bobpappas** [comment]
Woo hoo, it's working in Travis. @cixzhang, please review.

I just added comments referring to the Chrome performance bug.

File: spec/targetSpec.js [hunk 1]
```diff
@@ -1,19 +1,496 @@
...
+    var links = [max];
+    for (var i = 0; i < max; i++) {
+      links[i] = { from: getRandomInt(max),
+                    to: getRandomInt(max)
+                 }
+    }
+    return links;
+  }
+
+  beforeEach(function () {
```
Comment (holson1) [inline spec/targetSpec.js]: Can we move this to a separate file in a future PR?

**Pull request merged**